src/super_gradients/training/transforms/keypoints/keypoints_random_affine.py

```diff
@@ -0,0 +1,193 @@
+import random
+from typing import List, Union, Iterable
+
+import cv2
+import numpy as np
+
+from super_gradients.common.object_names import Transforms
+from super_gradients.common.registry.registry import register_transform
+from super_gradients.training.datasets.data_formats.bbox_formats.xywh import xywh_to_xyxy, xyxy_to_xywh
+from super_gradients.training.samples import PoseEstimationSample
+from .abstract_keypoints_transform import AbstractKeypointTransform
+
+
+@register_transform(Transforms.KeypointsRandomAffineTransform)
+class KeypointsRandomAffineTransform(AbstractKeypointTransform):
+    """
+    Apply random affine transform to image, mask and joints.
+    """
+
+    def __init__(
+        self,
+        max_rotation: float,
+        min_scale: float,
+        max_scale: float,
+        max_translate: float,
+        image_pad_value: int,
+        mask_pad_value: float,
+        interpolation_mode: Union[int, List[int]] = cv2.INTER_LINEAR,
+        prob: float = 0.5,
+    ):
+        """
+
+        :param max_rotation: Max rotation angle in degrees
+        :param min_scale: Lower bound for the scale change. For +- 20% size jitter this should be 0.8
+        :param max_scale: Lower bound for the scale change. For +- 20% size jitter this should be 1.2
+        :param max_translate: Max translation offset in percents of image size
+        :param interpolation_mode: A constant integer or list of integers, specifying the interpolation mode to use.
+        Possible values for interpolation_mode:
+          cv2.INTER_NEAREST = 0,
+          cv2.INTER_LINEAR = 1,
+          cv2.INTER_CUBIC = 2,
+          cv2.INTER_AREA = 3,
+          cv2.INTER_LANCZOS4 = 4
+        To use random interpolation modes on each call, set interpolation_mode = (0,1,2,3,4)
+        """
+        super().__init__()
+
+        self.max_rotation = max_rotation
+        self.min_scale = min_scale
+        self.max_scale = max_scale
+        self.max_translate = max_translate
+        self.image_pad_value = image_pad_value
+        self.mask_pad_value = mask_pad_value
+        self.prob = prob
+        self.interpolation_mode = tuple(interpolation_mode) if isinstance(interpolation_mode, Iterable) else (interpolation_mode,)
+
+    def __repr__(self):
+        return (
+            self.__class__.__name__ + f"(max_rotation={self.max_rotation}, "
+            f"min_scale={self.min_scale}, "
+            f"max_scale={self.max_scale}, "
+            f"max_translate={self.max_translate}, "
+            f"image_pad_value={self.image_pad_value}, "
+            f"mask_pad_value={self.mask_pad_value}, "
+            f"prob={self.prob})"
+        )
+
+    def _get_affine_matrix(self, img, angle, scale, dx, dy):
+        """
+
+        :param center: (x,y)
+        :param scale:
+        :param output_size: (rows, cols)
+        :param rot:
+        :return:
```

Comment:
docstrings + typing

Reply:
Fixed in 88cef73f0a3c26f9e158589c0d36ef541787eaee